test/foundry/utils/DeployedSetUp.t.sol

```diff
@@ -0,0 +1,69 @@
+// // SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.13;
+
+import {Test, Vm, console} from "forge-std/Test.sol";
+import {Constants} from "./Constants.sol";
+
+contract DeployedSetUp is Test {
+  uint256 constant notValidForkId = 9999999;
+  uint256 forkId = notValidForkId;
+  address holographInterfacesDeployed;
+  address holographDeployed;
+  address holographERC721Deployed;
+  address cxipERC721ProxyDeployed;
+  address cxipERC721Deployed;
+  address erc20MockDeployed;
+  address holographBridgeDeployed;
+  address holographBridgeProxyDeployed;
+  address holographERC20Deployed;
+  address holographFactoryDeployed;
+  address holographFactoryProxyDeployed;
+  address holographGenesisDeployed;
+  address holographOperatorDeployed;
+  address holographOperatorProxyDeployed;
+  address holographRegistryDeployed;
+  address holographRegistryProxyDeployed;
+  address holographTreasuryDeployed;
+  address holographTreasuryProxyDeployed;
+  address hTokenDeployed;
+  address mockERC721ReceiverDeployed;
+  address holographRoyaltiesDeployed;
+  address sampleERC20Deployed;
+  address sampleERC721Deployed;
+  address sampleERC721Deployed2;
+
+  function init(uint256 _forkId) public {
+    require(_forkId != notValidForkId, "DeployedSetUp: _forkId cannot be 9999999");
+    forkId = _forkId;
+  }
+
+  function setUp() public virtual {
+    require(forkId != notValidForkId, "DeployedSetUp: you need init first, forkId cannot be 9999999");
+    holographInterfacesDeployed = Constants.getHolographInterfaces();
+    holographDeployed = Constants.getHolograph();
+    holographERC721Deployed = Constants.getHolographERC721();
+    if (vm.activeFork() == forkId) cxipERC721ProxyDeployed = Constants.getCxipERC721Proxy_L2();
```

Comment:
question: why are we selecting based on activeFork? shouldn't this be checking the block.chainId?

Reply:
Since the active fork is the one running the tests, it can jump from one network to another.